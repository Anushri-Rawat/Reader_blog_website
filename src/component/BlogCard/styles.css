.card-heading {
  line-height: 24px !important;
  font-size: 24px !important;
  font-weight: 700 !important;
  text-overflow: ellipsis !important;
  color: rgba(41, 41, 41, 1) !important;
  padding-bottom: 8px !important;
  margin-bottom: 0 !important;
}
@media (max-width: 425px) {
  .user-info span,
  .user-info p {
    font-size: 14px;
  }
}
