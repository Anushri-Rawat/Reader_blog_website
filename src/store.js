import { createStore, applyMiddleware } from "redux";
import thunk from "redux-thunk";
import { composeWithDevTools } from "redux-devtools-extension";
import posts from "./reducers";

const initialState = {
  loading: false,
  posts: JSON.parse(localStorage.getItem("posts")) || [],
};

const store = () =>
  createStore(posts, initialState, composeWithDevTools(applyMiddleware(thunk)));

export default store;
