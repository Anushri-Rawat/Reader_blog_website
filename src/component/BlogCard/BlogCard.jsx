import React from "react";
import userData from "../../utils";
import { useDispatch } from "react-redux";
import "./styles.css";

const BlogCard = ({ data }) => {
  const dispatch = useDispatch();

  if (!data || Object.keys(data).length === 0)
    return <p data-testid="card">No data in blog to post</p>;

  return (
    <div
      className="border-bottom"
      style={{ marginTop: "24px" }}
      data-testid="card"
    >
      <div
        className="d-flex align-items-center gap-3"
        style={{ marginBottom: "1rem" }}
      >
        <img
          className="rounded-circle"
          src={userData[data.userId - 1].image}
          alt="user-img"
          style={{ width: "48px", height: "48px" }}
        />
        <div className="d-flex flex-column user-info">
          <div className="d-flex gap-1 align-items-center">
            <span>{userData[data.userId - 1].name}</span>
            <span className="fs-4 fw-bolder mb-2 d-none d-sm-block">.</span>
            <span className="d-none d-sm-block">
              @{userData[data.userId - 1].username}
            </span>
            <span className="fs-4 fw-bolder mb-2">.</span>
            <span>May 13</span>
          </div>
          <p className="mb-0">{userData[data.userId - 1].catchPhrase}</p>
        </div>
      </div>
      <div className="d-flex flex-column">
        <h2 className="card-heading">{data.title}</h2>
        <p className="mb-0">{data.body}</p>
      </div>
      <div
        style={{ padding: "32px 0" }}
        className="d-flex justify-content-between"
      >
        <div className="d-flex gap-3 align-items-center">
          <button
            className="border-0"
            style={{ padding: "2px 20px", borderRadius: "100px" }}
          >
            React
          </button>
          <span className="text-secondary">4 min read</span>
        </div>
        <button
          className="btn btn-outline-secondary"
          onClick={() => dispatch({ type: "DELETE_POST", payload: data })}
        >
          Delete
        </button>
      </div>
    </div>
  );
};

export default BlogCard;
