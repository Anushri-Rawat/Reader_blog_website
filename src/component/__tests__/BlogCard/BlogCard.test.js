import { render as rtlRender, screen, cleanup } from "@testing-library/react";
import BlogCard from "../../BlogCard/BlogCard";
import store from "../../../store";
import { Provider } from "react-redux";
import "@testing-library/jest-dom";
import renderer from "react-test-renderer";

afterEach(() => {
  cleanup();
});

const render = (component) =>
  rtlRender(<Provider store={store()}>{component}</Provider>);

const post = {
  body: "quia et suscipit\nsuscipit recusandae consequuntur expedita et cum\nreprehenderit molestiae ut ut quas totam\nnostrum rerum est autem sunt rem eveniet architecto",
  id: 1,
  title:
    "sunt aut facere repellat provident occaecati excepturi optio reprehenderit",
  userId: 1,
};

test("should render blogCard component with data", () => {
  render(<BlogCard data={post} />);
  const blogCardElement = screen.getByTestId("card");
  expect(blogCardElement).toBeInTheDocument();
});

test("should render blogCard component with no data", () => {
  render(<BlogCard data={{}} />);
  const blogCardElement = screen.getByTestId("card");
  expect(blogCardElement).toBeInTheDocument();
});

test("matches snapshot", () => {
  const tree = renderer
    .create(
      <Provider store={store()}>
        <BlogCard data={post} />
      </Provider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
