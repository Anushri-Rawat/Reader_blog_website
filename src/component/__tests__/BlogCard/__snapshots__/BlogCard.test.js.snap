// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="border-bottom"
  data-testid="card"
  style={
    Object {
      "marginTop": "24px",
    }
  }
>
  <div
    className="d-flex align-items-center gap-3"
    style={
      Object {
        "marginBottom": "1rem",
      }
    }
  >
    <img
      alt="user-img"
      className="rounded-circle"
      src="/assets/1.jpeg"
      style={
        Object {
          "height": "48px",
          "width": "48px",
        }
      }
    />
    <div
      className="d-flex flex-column user-info"
    >
      <div
        className="d-flex gap-1 align-items-center"
      >
        <span>
          Leanne Graham
        </span>
        <span
          className="fs-4 fw-bolder mb-2 d-none d-sm-block"
        >
          .
        </span>
        <span
          className="d-none d-sm-block"
        >
          @
          Bret
        </span>
        <span
          className="fs-4 fw-bolder mb-2"
        >
          .
        </span>
        <span>
          May 13
        </span>
      </div>
      <p
        className="mb-0"
      >
        Multi-layered client-server neural-net
      </p>
    </div>
  </div>
  <div
    className="d-flex flex-column"
  >
    <h2
      className="card-heading"
    >
      sunt aut facere repellat provident occaecati excepturi optio reprehenderit
    </h2>
    <p
      className="mb-0"
    >
      quia et suscipit
suscipit recusandae consequuntur expedita et cum
reprehenderit molestiae ut ut quas totam
nostrum rerum est autem sunt rem eveniet architecto
    </p>
  </div>
  <div
    className="d-flex justify-content-between"
    style={
      Object {
        "padding": "32px 0",
      }
    }
  >
    <div
      className="d-flex gap-3 align-items-center"
    >
      <button
        className="border-0"
        style={
          Object {
            "borderRadius": "100px",
            "padding": "2px 20px",
          }
        }
      >
        React
      </button>
      <span
        className="text-secondary"
      >
        4 min read
      </span>
    </div>
    <button
      className="btn btn-outline-secondary"
      onClick={[Function]}
    >
      Delete
    </button>
  </div>
</div>
`;
